src/api/routes/rosetta/construction.ts

```diff
@@ -226,47 +267,70 @@ export function createRosettaConstructionRouter(db: DataStore, chainId: ChainID)
     }
     const txSize: number = options.size;
 
-    const recipientAddress = options.token_transfer_recipient_address;
-    if (options?.decimals !== RosettaConstants.decimals) {
-      res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidCurrencyDecimals]);
-      return;
+    let response: RosettaConstructionMetadataResponse;
+    switch (options.type) {
+      case 'token_transfer':
+        const recipientAddress = options.token_transfer_recipient_address;
+        if (options?.decimals !== RosettaConstants.decimals) {
+          res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidCurrencyDecimals]);
+          return;
+        }
+    
+        if (recipientAddress == null || !isValidC32Address(recipientAddress)) {
+          res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidRecipient]);
+          return;
+        }
+        break;
+      case 'stacking':
+        // Getting stacking info
+        const poxInfo = await new StacksCoreRpcClient().getPox();
+        const coreInfo = await new StacksCoreRpcClient().getInfo();
+        const contractInfo = poxInfo.contract_id.split('.')
+        options.contract_address = contractInfo[0];
+        options.contract_name = contractInfo[1];
+        // Adding 3 blocks to provide a buffer for transaction to confirm
+        options.burn_block_height = coreInfo.burn_block_height + 3;
+        break;
```

Comment:
We could also here the values from poxInfo:
min_amount_ustx, prepare_cycle_length, reward_cycle_length

What do you think ?

Reply:
If we want to do some kind of pre-checks those could be useful but I assume that when starting the process of crafting a  transaction they already have those informations.

I would say keep the answer with only what is strictly needed.

Reply:
Right, let's keep it this way